src/locked_mem.rs

```diff
@@ -47,17 +47,25 @@ impl LockedArray {
 }
 
 impl From<Vec<u8>> for LockedArray {
-    fn from(s: Vec<u8>) -> Self {
+    fn from(mut s: Vec<u8>) -> Self {
         let mut out = Self::new(s.len()).unwrap();
         out.lock().copy_from_slice(s.as_slice());
+
+        // clear the input buffer
+        s.fill(0);
```

Comment:
Apparently there's a chance the optimizer will just delete something like this if the data is about to be freed anyways...

```suggestion
    fn from(mut s: Vec<u8>) -> Self {
        let s = zeroize::Zeroizing::new(s);
        let mut out = Self::new(s.len()).unwrap();
        out.lock().copy_from_slice(s.as_slice());
```

Reply:
The article linked from zeroize is interesting. I hadn't thought about this being optimized away but it makes sense.